sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -734,4 +734,54 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
         df1.groupBy("j").agg(max("k")))
     }
   }
+
+  //  A test with a partition where the number of files in the partition is
+  //  large. tests for the condition where the serialization of such a task may result in a stack
+  //  overflow if the files list is stored in a recursive data structure
+  //  This test is ignored because it takes long to run (~3 min)
+  ignore("SPARK-27100 stack overflow: read data with large partitions") {
```

Comment:
trigger the test and see whether it passes in 2.4?

Reply:
@parthchandra can you confirm this? Thanks.

Reply:
Yes. The test passes in 2.4 for both bucketing with/without hive.